snooty/parser.py

```diff
@@ -940,6 +943,18 @@ def validate_and_add_asset(
                 CannotOpenFile(Path(image_argument), err.strerror, line)
             )
 
+    def validate_relative_url(self, url_argument: str, line: int) -> None:
+        """Validate relative URL points to page within current docs site.
+        URLs can be of the form /foo, foo, /foo/"""
+        relative_txt_path = Path(url_argument).with_suffix(".txt")
```

Comment:
@rayangler I have one outstanding concern here which is that the comment `URLs can be of the form /foo, foo, /foo/` is not entirely correct. Because `Path` automatically strips any trailing slashes, if a writer put `/foo//` as the card URL, the parser would not complain. However, this would cause issues on the frontend as our users could potentially be routed to `<url>//`, which would always 404.

adding `url_argument.endswith("//")` to the if statement doesn't feel elegant to me, especially since the error could report an open file error on a valid path (since `absolute` will handle double slashes)

Any thoughts on a clean way to handle the double slash?

Reply:
I'm unsure of the best way for this particular case. Do you have any expert Python knowledge on this @i80and ? I assume you could write a regex to check that there's no more than one consecutive `/` in the string at a time, but I'm not sure

Reply:
I added the regex check to enforce the invariant specified in the comment & align with the expectations of the frontend